### PR TITLE
feat: couldn’t find environments message

### PR DESCRIPTION
**Pull request by meniRoy**
show message to the user when couldn’t find environments for current branch 
show message to the user when couldn’t find current branch 
Screenshot

https://github.com/env0/vscode-env0/assets/33559192/ae4cd7e2-28cc-4494-b7f6-5dd03628e0ae

### CODE REVIEW

**Comment by meniRoy** [comment]
[show "couldn't find environments" if none was found](https://trello.com/c/IKpR19Pt/10423-show-couldnt-find-environments-if-none-was-found)

File: src/test/unit/extension.test.ts [hunk 1]
```diff
@@ -31,8 +31,7 @@ describe("extension", () => {
     it("should display simple loading message when branch name is yet to load", async () => {
```
Comment (meniRoy) [inline src/test/unit/extension.test.ts:31]: this test i an old unit test runnig with jest that will be refactored to mocha in unit tests ticket

File: src/state-manager.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      return;
+    }
+    if (!this._isLoggedIn) {
+      this.environmentsView.message =
+        "you are logged out. in order to log in, run the command 'env0.login'";
+    } else if (this._failedToGetBranch) {
+      this.environmentsView.message = "Could not find current git branch.";
+    } else if (this._noEnvironment) {
+      if (this._currentBranch) {
+        this.environmentsView.message = `couldn’t find environments associated with ${this._currentBranch}, please create one`;
```
Comment (GiliFaroEnv0) [inline src/state-manager.ts:70]:
```suggestion
        this.environmentsView.message = `couldn’t find environments associated with current branch "${this._currentBranch}"`;
```

Comment (GiliFaroEnv0) [reply src/state-manager.ts:70]: no need to tell the customer to "create one",
you can maybe add some info on **how** to create one

File: src/state-manager.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      } else {
+        this.environmentsView.message = `couldn’t find environments associated with current branch, please create one`;
```
Comment (GiliFaroEnv0) [inline src/state-manager.ts:72]: no sure how this can happen, if you don't know the `currentBranch` by what are you searching ?

Comment (GiliFaroEnv0) [reply src/state-manager.ts:72]: and again no need to tell the customer to "create one",
you can maybe add some info on how to create one

File: src/state-manager.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      }
+    } else if (this._isLoading) {
+      if (this._currentBranch) {
+        this.environmentsView.message = `loading environments for ${this._currentBranch}...`;
+      } else {
+        this.environmentsView.message = "loading environments...";
+      }
```
Comment (GiliFaroEnv0) [inline src/state-manager.ts:79]:
```suggestion
      if (this._currentBranch) {
        this.environmentsView.message = `loading environments for ${this._currentBranch ? `for branch ${this._currentBranch}` : ''}...`;
      }
```

File: src/env0-environments-provider.ts [hunk 5]
```diff
@@ -106,13 +107,22 @@ export class Env0EnvironmentsProvider
...
-      if (this.shouldUpdate(newEnvironments) && this.isRefreshing) {
+      // we need to avoid taking any action in case user logged out while we were waiting for the response
+      if (!stateManager.isLoggedIn) {
+        return;
+      }
+      if (newEnvironments.length === 0) {
+        stateManager.setNoEnvironment(true);
+      } else {
+        stateManager.setNoEnvironment(false);
+      }
```
Comment (GiliFaroEnv0) [inline src/env0-environments-provider.ts:124]:
```suggestion
        stateManager.setNoEnvironment(newEnvironments.length === 0);
      }```

Comment (GiliFaroEnv0) [reply src/env0-environments-provider.ts:124]:
```suggestion
       stateManager.setNoEnvironment(_.isEmpty(newEnvironments));
```
is even better

File: src/get-environments.ts [hunk 6]
```diff
@@ -45,6 +46,12 @@ export async function getEnvironmentsForBranch() {
 
   if (environments.length > 0) {
     const { currentBranch, repository } = getGitRepoAndBranch();
+    if (!currentBranch || !repository) {
```
Comment (GiliFaroEnv0) [inline src/get-environments.ts:49]: add tests

File: src/extension.ts [hunk 7]
```diff
@@ -34,14 +35,19 @@ export const setContextShowLoginMessage = async (value: boolean) => {
 };
```
Comment (GiliFaroEnv0) [inline src/extension.ts:36]: this is the main file make sure to add tests here

File: src/state-manager.ts [hunk 8]
```diff
@@ -0,0 +1,85 @@
+import * as vscode from "vscode";
+import { Environment } from "./env0-environments-provider";
+
+export class StateManager {
```
Comment (GiliFaroEnv0) [inline src/state-manager.ts:4]: BAD NAME, when ever you have "manager" in the name should rise a warning.
maybe
`extension-state`
`extension-state-service`

File: src/state-manager.ts [hunk 9]
```diff
@@ -0,0 +1,85 @@
+import * as vscode from "vscode";
```
Comment (GiliFaroEnv0) [inline src/state-manager.ts:1]: this is also going to be a main logic file so consider adding tests to it.
I know now the logic is really "simple & stupid" but still (not a priority)

**Review by GiliFaroEnv0 — COMMENTED** [review]
Generally look good 😄 
left some comments

File: src/state-manager.ts (hunk 2 above)
Comment (meniRoy) [reply src/state-manager.ts]: 4bb1f1bb2f8546db218e896a6c76546849b531ec
We wanted to add a link to the documentation in those messages. However, due to certain limitations in VSCode, it's a bit complicated. Therefore, we've decided to delay adding the links until Phase 2. (See more [here](https://env0.slack.com/archives/C041FQKHX2T/p1692810979393739).)"

File: src/state-manager.ts (hunk 4 above)
Comment (meniRoy) [reply src/state-manager.ts]: 1ceadc39ab3bd80a5dbc80f9245368fbbda1ee05

File: src/env0-environments-provider.ts (hunk 5 above)
Comment (meniRoy) [reply src/env0-environments-provider.ts]: a644f33e6ec87b1b6d421e178c4858d48d44fb2f

File: src/state-manager.ts (hunk 8 above)
Comment (meniRoy) [reply src/state-manager.ts]: 43cf3f6787a0d6fc61607d41cb77a7fffff10a3b

File: src/extension.ts (hunk 7 above)
Comment (meniRoy) [reply src/extension.ts:36]: Most of the flows on this file are already tested in our integration tests
but maybe we should consider adding tests for those messages.

File: src/get-environments.ts (hunk 6 above)
Comment (meniRoy) [reply src/get-environments.ts:49]: fcdbf0ff3057ded31d46c33ae0c91d502de717e1

File: src/state-manager.ts (hunk 9 above)
Comment (meniRoy) [reply src/state-manager.ts]: fcdbf0ff3057ded31d46c33ae0c91d502de717e1

**Review by GiliFaroEnv0 — APPROVED** [review]
Good work 😄

**Pull request merged**